### ISSUE TITLE
[JENKINS-6756] Loading Overlay remains visible after page loading has finished

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#4251]
<p>When opening the configuration of a Job, the "Loading" overlay remains visible after the page has finished loading (see screenshots)<br/>
In Firefox there is a workaround as the overlay window is transparent, I can navigate with tab to the input field and can do the changes using keyboard only.<br/>
In Internet Explorer, the overlay window is grey, without any transparency, so it is not visible where to navigate to.</p>

<p>It seems as if this happens with a Maven2 project. There are multiple Free-Form (Ant) Projects and one Maven2 project. And only the Maven2 project shows this behavior.</p>

**Attachments:**

- [Firefox.jpg](https://issues.jenkins.io/secure/attachment/19508/Firefox.jpg)
- [InternetExplorer.jpg](https://issues.jenkins.io/secure/attachment/19509/InternetExplorer.jpg)
- [JavaScriptError.jpg](https://issues.jenkins.io/secure/attachment/19512/JavaScriptError.jpg)
- [JavaScriptError.jpg](https://issues.jenkins.io/secure/attachment/19511/JavaScriptError.jpg)
- [config.xml](https://issues.jenkins.io/secure/attachment/19667/config.xml)
- [configure.htm](https://issues.jenkins.io/secure/attachment/24342/configure.htm)
- [hudson-behavior.js](https://issues.jenkins.io/secure/attachment/19850/hudson-behavior.js)

---
<details><summary><i>Originally reported by <a title="Michael Rumpf" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mirumpf">mirumpf</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-6756" target="_blank">Loading Overlay remains visible after page loading has finished</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Alan Harder" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>
<li><b>reported by</b>: mirumpf
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2010-11-17T10:46:42+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 4
<li><b>imported</b>: 2025-11-06
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=leegee">leegee</a>:</i>
<p>Job config.xml for job with stuck "LOADING" overlay.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=leegee">leegee</a>:</i>
<p>I do not experience the same problem through FireFox, just in IE (v7 - corporate client's choice)</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=leegee">leegee</a>:</i>
<p>In the latest Mozilla, I have found at times (which times?) that the overlay appears with an alpha level of about 30%: the form can be submitted, and produces the error "The server encountered an internal error () that prevented it from fulfilling this request"</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>:</i>
<p>leegee, any updates?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ofarrell">ofarrell</a>:</i>
<p>mindless,</p>

<p>I had the same issue with my instance. Like you, I'm running Hudson in a Tomcat 6 installation. I was able to fix the problem by shutting down the server and cleaning out the ${CATALINA_HOME}/temp and ${CATALINA_HOME}/work directories. Now I'm back up and running. While not a fix, it's at least a workaround.</p>

<p>My suspicion is that this stems from containers persisting session data across restarts. The URLs for static content definitely make it seem as if there is some URL rewriting magic going on some sort of UID.</p>

<p>What would happen in a scenario where an application container loads unexpired session data from disk? Does Hudson have the awareness that a session may be new to it, but not the container? Someone familiar with the source may be able to shed light on how Hudson it integrates as a session listener (if at all).</p>

<p>~Owen</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=wolfs">wolfs</a>:</i>
<p>We have a similar Error. Clicking configure on any job yields the LOADING-Overlay which doesn't go away for a very long time (> 60 sec). Then it finishes and we can edit the configuration regularly. Firebug does not report any JS-Errors. We are running Hudson in a tomcat instance.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=wolfs">wolfs</a>:</i>
<p>Our Problem is solved. The reason was a wrong password for the artifactory server set in the artifactory plugin.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>:</i>
<p>I'm reclosing this issue.. from the additional comments here I don't see anything further I can do.  If anyone still sees a problem, please collect the javascript error reported by the browser and the HTML source of the page, and include these in a new issue, thanks.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kevinp2">kevinp2</a>:</i>
<p>Issue has reappeared in Hudson 1.380 using IE 8.</p>

<p>Can we get rid of the Loading overlay altogether?  It adds little value and makes Hudson unusable when it malfunctions.  Please get rid of it or reduce it to a small "Loading ... Please wait" message in red at the top of the page.</p>

<p>Javascript error message:<br/>
Webpage error details</p>

<p>User Agent: Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 6.0; Trident/4.0; SLCC1; .NET CLR 2.0.50727; InfoPath.2; .NET CLR 3.5.30729; .NET CLR 3.0.30729; OfficeLiveConnector.1.4; OfficeLivePatch.1.3; .NET CLR 1.1.4322; MS-RTC LM 8)<br/>
Timestamp: Tue, 12 Oct 2010 17:17:28 UTC</p>

<p>Message: Object doesn't support this property or method<br/>
Line: 1031<br/>
Char: 13<br/>
Code: 0<br/>
URI: <a href="http://auidc84.somecompany.com:8888/static/8caf2d87/scripts/hudson-behavior.js" class="external-link" target="_blank" rel="nofollow noopener">http://auidc84.somecompany.com:8888/static/8caf2d87/scripts/hudson-behavior.js</a></p>

<p>Installed Plug-ins:<br/>
Audit Trail version 1.4<br/>
Blame Upstream Committers 1.2<br/>
Copy Artifact Plugin 1.10<br/>
Hudson CVS Plug-in 1.2<br/>
Hudson disk-usage plugin 0.11<br/>
Downstream build view 1.4<br/>
Hudson Email Extension Plugin 2.8<br/>
Green Balls 1.6<br/>
LDAP Email Plugin 0.6<br/>
Maven Integration plugin 1.380<br/>
Hudson Parameterized Trigger plugin 2.4<br/>
Hudson Support Subscription Notification Plugin 1.2<br/>
Hudson SSH Slaves plugin 0.13   <br/>
Hudson Subversion Plug-in 1.17</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kevinp2">kevinp2</a>:</i>
<p>According to IE8 Javascript error message, the error occurs in file hudson-behavior.js (attached) in function refillOnChange at line 1031 which is:</p>

<p>c.addEventListener("change",h,false);</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kevinp2">kevinp2</a>:</i>
<p>An update:  This issue seems related to the Copy Artifacts plugin for some reason.  I removed the plugin from this job and the Javascript error went away.  The Copy Artifacts plugin seems to be having some major issues in Hudson 1.380</p>

<p>However, in any event:<br/>
If a plugin misbehaves, the entire job should not be rendered unusable.<br/>
The loading overlay should not cover the entire screen and render the job uneditable.</p>

<p>So please change Hudson to be more robust.  Thanks <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke">kohsuke</a>:</i>
<p>I just had an opportunity to see this problem first hand. Here's the report of the details.</p>

<p>This user had a truncated YUI menu-min.js in his browser cache. Because of the way we serve JavaScript files, once this happens, the file won't be reloaded by the browser unless you do a deep refresh (SHIFT + reload icon, for example.)</p>

<p>menu-min.js fails to load, which in turn results in additional JavaScript problems as some of the forms try to use the menu. Presumably then hudson-behaviours.js fail to apply the rest, and the overlay remains.</p>

<p>It wasn't clear what caused the truncated menu-min.js file to be served in the first place. The container was Winstone, but a quick test based on wget worked, and shift+reload did fix the problem.</p>

<p>I think what's happening here is that every time some JavaScript problem happens in the config page, the failure mode is that 'OVERLAY' being stuck.</p>

<p>I wonder if it's possible to repoprt the JavaScript problem in a more explicit way so that we can better understand the root cause.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=szabobakter">szabobakter</a>:</i>
<p>I had the same issue, and i can affirm that the bug is related to the Copy Artifact plugin.<br/>
I get the error only in IE8, and only after i added a Copy Artifact build step to a project (freestyle or maven project with m2-extra-steps).<br/>
I'm running the last 1.383 Hudson version under Tomcat 5.5 and the error still remains.<br/>
The js error is "Object doesn't support this property or method" in hudson-behavior.js line 1031 char 13 as noted above, and the SHIFT + reload doesn't help.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rherrick">rherrick</a>:</i>
<p>This was previously happening to me when I copied the CopyArtifact configuration into a job's config.xml, but is now doing it whenever I add the copy artifact step to my job configuration.  I have ALMOST exactly the same error as szabobakter, except that it's line 1030 instead of 1031.  I have a slightly different version, so that probably accounts for that.</p>

<p>Hudson v1.377<br/>
JBoss 5.1.0.GA<br/>
JDK 6<br/>
IE 8 (regular and in compatibility view).</p>

<p>If I view the page in Firefox 3.6.10, things appear properly.  There is still a JS error on the page, but a <em>different</em> JS error and the Loading overlay goes away.</p>

<p>Let me know if I can provide you with any technical data to help resolve this issue.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mponomarenko">mponomarenko</a>:</i>
<p>happens for me also. Hudson 1.384. happens in IE, Crhome and Opera. When I scroll below "loading" overay and and sumit page I get "satus code 400:Exception: This page expects a form submission<br/>
Stacktrace:<br/>
(none)"</p>

<p>Here is original error from opera (most derscriptive one):</p>

<p>JavaScript - <a href="http://localhost:8383/job/deploy-scripts/configure" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:8383/job/deploy-scripts/configure</a><br/>
Inline script compilation<br/>
Syntax error at line 2062 while loading:<br/>
stributed under the <br/>
--------------------^<br/>
in multiline comment: unexpected end of script<br/>
JavaScript - <a href="http://localhost:8383/job/deploy-scripts/configure" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:8383/job/deploy-scripts/configure</a><br/>
Inline script compilation<br/>
Syntax error at line 12 while loading:<br/>
)&&(L==H)){K=true;}}<br/>
--------------------^<br/>
expected expression, got <end of file><br/>
JavaScript - <a href="http://localhost:8383/job/deploy-scripts/configure" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:8383/job/deploy-scripts/configure</a></p>

<p>Uncaught exception: Error: YAHOO.lang.extend failed, please check that all dependencies are included.<br/>
Error thrown at line 7, column 5347 in <anonymous function: extend>(D, E, C) in <a href="http://localhost:8383/static/66ffcbeb/scripts/yui/yahoo/yahoo-min.js:" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:8383/static/66ffcbeb/scripts/yui/yahoo/yahoo-min.js:</a><br/>
    throw new Error("YAHOO.lang.extend failed, please check that "+"all dependencies are included.");<br/>
called from line 7, column 8881 in <anonymous function>() in <a href="http://localhost:8383/static/66ffcbeb/scripts/yui/menu/menu-min.js:" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:8383/static/66ffcbeb/scripts/yui/menu/menu-min.js:</a><br/>
    YAHOO.lang.extend(F,B,{CSS_CLASS_NAME:"yuimenu",ITEM_TYPE:null,GROUP_TITLE_TAG_NAME:"h6",OFF_SCREEN_POSITION:<span class="error">&#91;-10000,-10000&#93;</span>,_nHideDelayId:null,_nShowDelayId:null,_nSubmenuHideDelayId:null,_nBodyScrollId:null,_bHideDelayEventHandlersAssigned:false,_bHandledMouseOverEvent:false,_bHandledMouseOutEvent:false,_aGroupTitleElements:null,_aItemGroups:null,_aListElements:null,_nCurrentMouseX:0,_bStopMouseEventHandlers:false,_sClassName:null,lazyLoad:false,itemData:null,activeItem:null,parent:null,srcElement:null,mouseOverEvent:null,mouseOutEvent:null,mouseDownEvent:null,mouseUpEvent:null,clickEvent:null,keyPressEvent:null,keyDownEvent:null,keyUpEvent:null,itemAddedEvent:null,itemRemovedEvent:null,init:function(P,O){this._aItemGroups=[];<br/>
called from line 7, column 3394 in <a href="http://localhost:8383/static/66ffcbeb/scripts/yui/menu/menu-min.js:" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:8383/static/66ffcbeb/scripts/yui/menu/menu-min.js:</a><br/>
    (function(){YAHOO.widget.Menu=function(O,N){if(N)</p>
{this.parent=N.parent;this.lazyLoad=N.lazyLoad||N.lazyload;this.itemData=N.itemData||N.itemdata;}
<p>YAHOO.widget.Menu.superclass.constructor.call(this,O,N);};function I(N){if(typeof N=="string"){return("dynamic,static".indexOf((N.toLowerCase()))!=-1);}}var C=YAHOO.util.Dom,M=YAHOO.util.Event,D=YAHOO.widget.Module,B=YAHOO.widget.Overlay,F=YAHOO.widget.Menu,K=YAHOO.widget.MenuManager,L=YAHOO.util.CustomEvent,E=YAHOO.lang,H=YAHOO.env.ua,G,A=</p>
{"MOUSE_OVER":"mouseover","MOUSE_OUT":"mouseout","MOUSE_DOWN":"mousedown","MOUSE_UP":"mouseup","CLICK":"click","KEY_PRESS":"keypress","KEY_DOWN":"keydown","KEY_UP":"keyup","FOCUS":"focus","BLUR":"blur","ITEM_ADDED":"itemAdded","ITEM_REMOVED":"itemRemoved"}
<p>,J={"VISIBLE":</p>
{key:"visible",value:false,validator:E.isBoolean}
<p>,"CONSTRAIN_TO_VIEWPORT":</p>
{key:"constraintoviewport",value:true,validator:E.isBoolean,supercedes:["iframe","x","y","xy"]}
<p>,"POSITION":</p>
{key:"position",value:"dynamic",validator:I,supercedes:["visible","iframe"]}
<p>,"SUBMENU_ALIGNMENT":</p>
{key:"submenualignment",value:["tl","tr"],suppressEvent:true}
<p>,"AUTO_SUBMENU_DISPLAY":</p>
{key:"autosubmenudisplay",value:true,validator:E.isBoolean,suppressEvent:true}
<p>,"SHOW_DELAY":</p>
{key:"showdelay",value:250,validator:E.isNumber,suppressEvent:true}
<p>,"HIDE_DELAY":</p>
{key:"hidedelay",value:0,validator:E.isNumber,suppressEvent:true}
<p>,"SUBMENU_HIDE_DELAY":</p>
{key:"submenuhidedelay",value:250,validator:E.isNumber,suppressEvent:true}
<p>,"CLICK_TO_HIDE":</p>
{key:"clicktohide",value:true,validator:E.isBoolean,suppressEvent:true}
<p>,"CONTAINER":</p>
{key:"container",suppressEvent:true}
<p>,"SCROLL_INCREMENT":</p>
{key:"scrollincrement",value:1,validator:E.isNumber,supercedes:["maxheight"],suppressEvent:true}
<p>,"MIN_SCROLL_HEIGHT":</p>
{key:"minscrollheight",value:90,validator:E.isNumber,supercedes:["maxheight"],suppressEvent:true}
<p>,"MAX_HEIGHT":</p>
{key:"maxheight",value:0,validator:E.isNumber,supercedes:["iframe"],suppressEvent:true}
<p>,"CLASS_NAME":</p>
{key:"classname",value:null,validator:E.isString,suppressEvent:true}
<p>,"DISABLED":{key:"disabled",value:false,validator:E.isBoolean,suppressEvent:true}};YAHOO.lang.extend(F,B,{CSS_CLASS_NAME:"yuimenu",ITEM_TYPE:null,GROUP_TITLE_TAG_NAME:"h6",OFF_SCREEN_POSITION:<span class="error">&#91;-10000,-10000&#93;</span>,_nHideDelayId:null,_nShowDelayId:null,_nSubmenuHideDelayId:null,_nBodyScrollId:null,_bHideDelayEventHandlersAssigned:false,_bHandledMouseOverEvent:false,_bHandledMouseOutEvent:false,_aGroupTitleElements:null,_aItemGroups:null,_aListElements:null,_nCurrentMouseX:0,_bStopMouseEventHandlers:false,_sClassName:null,lazyLoad:false,itemData:null,activeItem:null,parent:null,srcElement:null,mouseOverEvent:null,mouseOutEvent:null,mouseDownEvent:null,mouseUpEvent:null,clickEvent:null,keyPressEvent:null,keyDownEvent:null,keyUpEvent:null,itemAddedEvent:null,itemRemovedEvent:null,init:function(P,O)</p>
{this._aItemGroups=[];
JavaScript - http://localhost:8383/job/deploy-scripts/configure
Inline script compilation
Syntax error at line 785 while loading:
// can sandwitch 
--------^
expected '}
<p>', got <end of file><br/>
JavaScript - <a href="http://localhost:8383/job/deploy-scripts/configure" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:8383/job/deploy-scripts/configure</a></p>

<p>Uncaught exception: ReferenceError: Undefined variable: crumb<br/>
Error thrown at line 1, column 0 in <a href="http://localhost:8383/job/deploy-scripts/configure:" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:8383/job/deploy-scripts/configure:</a><br/>
    crumb.init("", "");<br/>
JavaScript - <a href="http://localhost:8383/job/deploy-scripts/configure" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:8383/job/deploy-scripts/configure</a></p>

<p>Uncaught exception: ReferenceError: Undefined variable: createSearchBox<br/>
Error thrown at line 1, column 0 in <a href="http://localhost:8383/job/deploy-scripts/configure:" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:8383/job/deploy-scripts/configure:</a><br/>
    createSearchBox("/job/deploy-scripts/search/");<br/>
JavaScript - <a href="http://localhost:8383/job/deploy-scripts/configure" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:8383/job/deploy-scripts/configure</a></p>

<p>Uncaught exception: ReferenceError: Undefined variable: updateBuildHistory<br/>
Error thrown at line 1, column 0 in <a href="http://localhost:8383/job/deploy-scripts/configure:" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:8383/job/deploy-scripts/configure:</a><br/>
    updateBuildHistory("/job/deploy-scripts/buildHistory/ajax",3);<br/>
JavaScript - <a href="http://localhost:8383/job/deploy-scripts/configure" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:8383/job/deploy-scripts/configure</a></p>

<p>Uncaught exception: TypeError: 'Behaviour.list._each' is not a function<br/>
Error thrown at line 58, column 8 in <anonymous function: applySubtree>(startNode) in <a href="http://localhost:8383/static/66ffcbeb/scripts/behavior.js:" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:8383/static/66ffcbeb/scripts/behavior.js:</a><br/>
    Behaviour.list._each(function(sheet) {<br/>
called from line 54, column 8 in <anonymous function: apply>() in <a href="http://localhost:8383/static/66ffcbeb/scripts/behavior.js:" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:8383/static/66ffcbeb/scripts/behavior.js:</a><br/>
    this.applySubtree(document);<br/>
called from line 49, column 3 in <anonymous function: start>() in <a href="http://localhost:8383/static/66ffcbeb/scripts/behavior.js:" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:8383/static/66ffcbeb/scripts/behavior.js:</a><br/>
    Behaviour.apply();</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in hudson<br/>
User: : mindless<br/>
Path:<br/>
 trunk/hudson/main/war/src/main/webapp/scripts/hudson-behavior.js<br/>
 trunk/www/changelog.html<br/>
<a href="http://jenkins-ci.org/commit/37011" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/37011</a><br/>
Log:<br/>
<span class="error">&#91;FIXED JENKINS-6756&#93;</span> fix js error in IE for f:select and f:combobox</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=salmira">salmira</a>:</i>
<p>Page source with <span class="error">&#91;LOADING&#93;</span> frame overlapping the completely loaded page. <br/>
Taken in <br/>
Jenkins ver. 1.530.<br/>
All plugins are up-to-date.<br/>
Browser: <br/>
Mozilla Firefox 23.0.1.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Originally related to: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-06-a2/issues?q=in%3Atitle%20JENKINS-8789">JENKINS-8789</a>]</i>
